### PR TITLE
Jro isr update

### PR DESCRIPTION
**Pull request by aburrell**
Fixed time bug, added location routines, and added xarray Madrigal support.

### CODE REVIEW

**Comment by aburrell** [comment]
Addresses issue #113 and contributes to issue #114

**Comment by rstoneback** [comment]
Awesome. Thanks Angeline.

I will take a look when I’m back in town.

Cheers,
Russell

On Nov 13, 2018, at 3:07 PM, Angeline Burrell <notifications@github.com<mailto:notifications@github.com>> wrote:


Fixed time bug, added location routines, and added xarray Madrigal support.

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/rstoneback/pysat/pull/118

Commit Summary

  *   solar local time
  *   Added coordinate conversion routines
  *   measurement location routine
  *   datetime bug
  *   Madrigal xarray support
  *   xarray for JRO ISR

File Changes

  *   M pysat/instruments/jro_isr.py<https://github.com/rstoneback/pysat/pull/118/files#diff-0> (47)
  *   M pysat/instruments/madrigal_methods.py<https://github.com/rstoneback/pysat/pull/118/files#diff-1> (58)
  *   M pysat/utils.py<https://github.com/rstoneback/pysat/pull/118/files#diff-2> (368)

Patch Links:

  *   https://github.com/rstoneback/pysat/pull/118.patch
  *   https://github.com/rstoneback/pysat/pull/118.diff

**Comment by aburrell** [comment]
The travis problem looks like an issue with Madrigal, since only one DMSP satellite had an issue in the Python 2 build, and the Python 3 build passed.

**Comment by aburrell** [comment]
JRO fixes led to massive modelling updates.  The model pull request #120 should be merged before this one.

File: pysat/instruments/jro_isr.py [hunk 1]
```diff
@@ -57,7 +57,7 @@
 jro_fname1 = 'jro{year:4d}{month:02d}{day:02d}'
 jro_fname2 = '.{version:03d}.hdf5'
 supported_tags = {ss:{'drifts':jro_fname1 + "drifts" + jro_fname2,
-                      'drifts_ave':jro_fname1 + "drifts_ave" + jro_fname2,
+                      'drifts_ave':jro_fname1 + "drifts_avg" + jro_fname2,
```
Comment (rstoneback) [inline pysat/instruments/jro_isr.py]: I'm curious why the testing didn't catch this typo.

Comment (aburrell) [reply pysat/instruments/jro_isr.py]: That's a good question.

Comment (rstoneback) [reply pysat/instruments/jro_isr.py]: Previously, download failures weren't strictly flagged as an issue. The unit test printed a note but moved on. In pull #143 the download check now has to pass or specifically be exempted.
For this particular issue, the download should have been ok but the typo in list files would report that there are no files, which is interpreted as a download failure (now).

**Pull request merged**